### PR TITLE
feat: Sync mode - least work approach

### PR DESCRIPTION
**Pull request by jakkab**
alternative approach to #4. Modify SyncPeriod to control intervals between forced reconciliations for each CR.

### CODE REVIEW

File: controllers/oauth2client_controller.go [hunk 1]
```diff
@@ -117,66 +117,68 @@ func (r *OAuth2ClientReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
...
-				return ctrl.Result{}, nil
-			}
+	if found {
+		//conclude reconciliation if the client exists and has not been updated
+		if oauth2client.Generation == oauth2client.Status.ObservedGeneration {
+			return ctrl.Result{}, nil
+		}
 
-			if updateErr := r.updateRegisteredOAuth2Client(ctx, &oauth2client, credentials); updateErr != nil {
+		if fetched.Owner != oauth2client.Name {
```
Comment (kubadz) [inline controllers/oauth2client_controller.go]: are you sure that it should be changed from `fetched.Owner != fmt.Sprintf("%s/%s", oauth2client.Name, oauth2client.Namespace)` to the current one? This field didn't change: https://github.com/ory/hydra-maester/blob/3422c11749c23918f0d8aa73508f5f515a0c08c2/api/v1alpha1/oauth2client_types.go#L181

Comment (jakkab) [reply controllers/oauth2client_controller.go]: I have no idea what happened here, I didn't intend to change that :O

**Comment by aeneasr** [comment]
LMK if good for 👀

**Comment by jakkab** [comment]
@aeneasr Ready for your review. Please bear in mind it’s not a fully fledged sync mode, it just allows the controller to reconcile an existing CR that has not been modified, but for some reason the corresponding Oauth2 client is not present in Hydra.

**Comment by aeneasr** [comment]
Looking good - thank you!

**Pull request merged**